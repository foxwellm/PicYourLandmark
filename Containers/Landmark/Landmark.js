import React, {Component} from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { Button, ButtonGroup, Icon } from 'react-native-elements';

export default class Landmark extends Component {
    constructor() {
        super()
        this.state = {
            collectedLandmarks: [
                {
                    id: 1,
                    name: 'Coors Field',
                    pointVal: 10,
                    visited: '4/1/2019'
                },
                {
                    id: 2,
                    name: 'Statue of Liberty',
                    pointVal: 10,
                    visited: '3/12/2019'
                },
                {
                    id: 3,
                    name: 'Grand Canyon',
                    pointVal: 10,
                    visited: '12/9/2018'
                }
            ]
        }
    }
    render() {
        return (
            <View style={styles.container}>
                {
                    this.state.collectedLandmarks.map((landmark => {
                        return(
                            <View key={landmark.id} style={styles.landmarkContainer}>
                                <Text>Landmark name: {landmark.name}</Text>
                                <Text>Point value: {landmark.pointVal}</Text>
                                <Text>Date visited: {landmark.visited}</Text>
                            </View>
                        )
                    }))
                }
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'column',
        left: '5%',
        width: '92%',
        padding: 8,
    },
    landmarkContainer: {
        padding: 10,
        top: -60
    }
});
