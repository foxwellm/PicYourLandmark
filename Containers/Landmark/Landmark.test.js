import React from 'react';
import { Landmark } from './Landmark';
import renderer from 'react-test-renderer';

describe('Landmark', () => {
  let mockLocation;
  test('renders correctly', () => {
    mockLocation = {
      name: "Great Lawn Park",
      description: "Beautiful Park",
      lat: 39.72386,
      lon: -104.88715,
      landmark_id: 6,
      photo_url: "www.myimage.com/2"
  }
    const tree = renderer.create(<Landmark key={1} landmark={mockLocation}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
