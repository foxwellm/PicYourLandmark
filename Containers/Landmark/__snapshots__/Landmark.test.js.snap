// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landmark renders correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "left": "5%",
      "padding": 8,
      "width": "92%",
    }
  }
>
  <View
    style={
      Object {
        "padding": 10,
        "top": -200,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "bottom": 15,
          "height": 200,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      COORS FIELD
       (
      4/1/2019
      )
    </Text>
    <View
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <View>
        <View
          left={-140}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          top={3}
          underlayColor="white"
        >
          <Text />
        </View>
      </View>
      <Text
        style={
          Object {
            "left": 30,
            "top": -15,
          }
        }
      >
        10
         gems
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "top": -200,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "bottom": 15,
          "height": 200,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      STATUE OF LIBERTY
       (
      3/12/2019
      )
    </Text>
    <View
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <View>
        <View
          left={-140}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          top={3}
          underlayColor="white"
        >
          <Text />
        </View>
      </View>
      <Text
        style={
          Object {
            "left": 30,
            "top": -15,
          }
        }
      >
        10
         gems
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "top": -200,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "bottom": 15,
          "height": 200,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      GRAND CANYON
       (
      12/9/2018
      )
    </Text>
    <View
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <View>
        <View
          left={-140}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
          top={3}
          underlayColor="white"
        >
          <Text />
        </View>
      </View>
      <Text
        style={
          Object {
            "left": 30,
            "top": -15,
          }
        }
      >
        10
         gems
      </Text>
    </View>
  </View>
</View>
`;
